docs/types/operators.rst

```diff
@@ -18,7 +18,9 @@ and the type of the operator's result:
 In case one of the operands is a :ref:`literal number <rational_literals>` it is first converted to its
 "mobile type", which is the smallest type that can hold the value
 (unsigned types of the same bit-width are considered "smaller" than the signed types).
-If both are literal numbers, the operation is computed with arbitrary precision.
+If both are literal numbers, the operation is computed with effectively unlimited precision in
+that the expression is evaluated to whatever precision is necessary so that no precision is lost
```

Comment:
Yes, this sounds better! Is there a way to avoid the third repetition of the word "precision"?

What I don't understand is why you say "when the result is used with a non-literal type". Is that really necessary (and correct)?

Reply:
"so that no precision is lost" --> "so that none is lost" fixes that.

I believe yes, it is necessary, because my reading of the docs (which I have not yet verified by reading a Solidity compiler implementation) is that these literal expressions are maintained in a computable but not computed form until the conversion to a "physical" type happens. I.e., that the "effectively unlimited precision" is a result not of actually storing the "unlimited precision" value, but rather, its fractional rational representation until that value can be computer *up to the limit of the destination type*.